### DISCUSSION TITLE
Nornir - Question about Merge vs. Replace

### DISCUSSION THREAD

**Discussion by gummidude** [nornir-automation/nornir#775] [Q&A]
Hi Guys,

I am using nornir together with the napalm connection plugin to modifiy access-lists and service-policys on Cisco IOS-XR ASR9k routers. 
Everything is working fine so far, but I encounted a behavior when updating Access lists that I want to change to my liking.

For example I want to add the following ACL:
```
ipv4 access-list githubtest
 permit ipv4 any 192.168.42.0/24
!
```
Using nornir+napalm it is no effort to add this acl to the router config. But if I want to rerun this 'add acl githubtest'-Task, the diff between running config and new config shows the following:
```
 ipv4 access-list githubtest
  10 permit ipv4 any 192.168.42.0/24
+ 20 permit ipv4 any 192.168.42.0/24
!
```
Meaning that it would add the already exisiting network to the same ACL. My expectation is that it is detected that this acl with the very same network already exisists on the router and no change is required. 

The workflow is as follows
1. $SomeOtherScript generates cisco-ios-xr config as String that should be applied to a specific router Alice, i.e. the new access-list named githubtest from above.
2. Now I create a config with 
```
r = task.run(task=template_string,
                 name="Config from interface-config",
                 template=interfaceConf #<-String from 1.
                 )
```

3. Now I want to apply this config to the router via
```
 result = task.run(task=napalm_configure,
            name="Loading Configuration on the device",
                      replace=False,
                      configuration=r.result,
                      dry_run=True
                      )
```

I guess it has to do something with the napalm run task replace=True/False parameter. But I could not figure it out, yet. 
When I set replace=True, nornir/napalm wants to completely erase the complete router config with the exception being the ACL o.O

I am using nornir 3.1.1 with the default inventory plugin, napalm 3.3.1, python3 3.7.3 and the router is running IOS-XR 6.6.3 64-bit.

Do you know how to fix this? 
Thanks in advance for your help!

With best regards,
Gummidude

**Comment by dbarrosop** [comment]
You should ask this question to the NAPALM folks.

**Reply by ktbyers** [reply]
Yeah, you should move this question to the NAPALM discussions or issues.

**Reply by gummidude** [reply]
Makes sense, thanks for the suggestion. Will move on to the NAPALM guys.
